phases/repo.go

```diff
@@ -2,112 +2,292 @@ package phases
 
 import (
 	"fmt"
+	"net/url"
+	"os"
 	"strings"
 
-	"github.com/bitrise-io/go-utils/command"
-	"github.com/bitrise-io/go-utils/errorutil"
 	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/goinp/goinp"
+	git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/storage/memory"
+)
+
+// RepoScheme is the type of the git repository protocol
+type RepoScheme int
+
+const (
+	// Invalid is an unsupported git repo scheme type
+	Invalid RepoScheme = iota
+	// HTTPS is the https git repo scheme type
+	HTTPS
+	// SSH is the ssh git repo scheme type
+	SSH
 )
 
 // RepoDetails encapsulates data needed to perform
 // repo registration related requests through the
 // Bitrise API
 type RepoDetails struct {
-	URL      string
-	Provider string
-	Owner    string
-	Slug     string
+	URL         string
+	Provider    string
+	Owner       string
+	Slug        string
+	Scheme      RepoScheme
+	SSHUsername string
 }
 
 type urlParts struct {
```

Comment:
I think this type is no longer needed. You only use it as a return value for `splitURL` and then use the fields to populate a `RepoDetails` object with extra props from the `*url.URL` passed to `splitURL`, which is readily available there.